lib/fs/filesystem.js

```diff
@@ -254,16 +270,40 @@ class FileSystem {
 
   createWriteStream(path, _options, _callback) {
     // https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options
-    const callback = (typeof _options === 'function') ? _options : _callback;
-    return operationWrapper(this.rest, 'upload', [path], (e, json) => {
+    let options;
+    let callback;
+
+    if (typeof _options === 'function') {
+      callback = _options;
+      options = {};
+    } else {
+      callback = _callback;
+      options = _options;
+    }
+
+    if (options && options.offset) {
+      options.headers = { Range: `bytes=${options.offset}-` };
+      delete options.offset;
+
+      if (options.timestamp) {
+        const ts = moment.unix(options.timestamp);
+        options.headers['If-Unmodified-since'] = ts.format('ddd, d M YYYY HH:mm:ss GMT');
```

Comment:
Any reason not to use ISO format?

Reply:
I was going by the spec of the `If-Unmodified-Since` header.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Unmodified-Since

Reply:
Would this need a capital `S` then?

Reply:
Yes. Done.